tests/subsys/rtio/rtio_api/src/test_rtio_api.c

```diff
@@ -954,6 +955,68 @@ ZTEST(rtio_api, test_rtio_await)
 	test_rtio_await_(&r_await0, &r_await1);
 }
 
+
+
+RTIO_DEFINE(r_callback_result, SQE_POOL_SIZE, CQE_POOL_SIZE);
+RTIO_IODEV_TEST_DEFINE(iodev_test_callback_result);
+static int callback_count;
+static int callback_result;
+static int expected_callback_result;
+
+void callback_update_data(struct rtio *r, const struct rtio_sqe *sqe,
+			  int result, void *arg0)
+{
+	_iodev_data_iodev_test_callback_result.result = expected_callback_result;
+	callback_count++;
+}
+
+void callback_stash_result(struct rtio *r, const struct rtio_sqe *sqe,
+			int result, void *arg0)
+{
+	callback_result = result;
+	callback_count++;
+}
+
+/*
+ * Ensure callbacks work as expected.
+ *
+ * 1. Callbacks always occur
+ * 2. The result code always contains the first error result
+ */
+ZTEST(rtio_api, test_rtio_callbacks)
```

Comment:
> * Callback operations are always called regardless of success/error status of previous submissions
>   in the chain.
> 

May we add a test-case to validate that callbacks in a chain of OPs will still be invoked even if the OP fails? As opposed to the current behavior where the rest of the chain is cancelled should this happen.